hail/src/main/scala/is/hail/rvd/AbstractRVDSpec.scala

```diff
@@ -189,16 +189,26 @@ object AbstractRVDSpec {
     val partitioner = specLeft.partitioner
 
     val extendedNewPartitioner = newPartitioner.map(_.extendKey(partitioner.kType))
-    val tmpPartitioner = extendedNewPartitioner match {
-      case Some(np) => np.intersect(partitioner)
-      case None => partitioner
+    val (parts, tmpPartitioner) = extendedNewPartitioner match {
+      case Some(np) =>
+        val tmpPart = np.intersect(partitioner)
+        assert(specLeft.key.nonEmpty)
+        val p = tmpPart.rangeBounds.map { b => specLeft.partFiles(partitioner.lowerBoundInterval(b)) }
+        (p, tmpPart)
+      case None =>
+        val partsBuilder = new BoxedArrayBuilder[String]()
+        val rbBuilder = new BoxedArrayBuilder[Interval]()
+        partitioner.rangeBounds.foreach { b =>
+          val idx = partitioner.lowerBoundInterval(b)
+          if (idx < partitioner.numPartitions) {
```

Comment:
This is confusing me. I don't see how this can ever be false: `idx` must be at most the index of `b`. If so, the None case looks equivalent to before. What was the intention here?

Reply:
This is not always true -- the test case I added here writes a table with a single partition whose bound is: `([0]-[0])`

`lowerBoundInterval` returns 1 in this case.

Reply:
Ah, I see. Thanks. I was assuming non-degenerate intervals.